apps/www/src/registry/default/plate-ui/inline-combobox.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { ReactNode, startTransition, useMemo, useState } from 'react';
+import {
+  Combobox,
+  ComboboxItem,
+  ComboboxPopover,
+  ComboboxProvider,
+  Portal,
+} from '@ariakit/react';
+import { cn } from '@udecode/cn';
+import {
+  BaseComboboxItemWithEditor,
+  matchWords,
+  useComboboxInput,
+  useHTMLInputCursorState,
+} from '@udecode/plate-combobox';
+import { insertText, moveSelection, useEditorRef } from '@udecode/plate-common';
+
+const comboboxItemClassName =
+  'relative flex h-9 select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none';
+const comboboxItemInteractiveClassName =
+  'cursor-pointer transition-colors hover:bg-accent hover:text-accent-foreground data-[active-item=true]:bg-accent data-[active-item=true]:text-accent-foreground';
+
+const defaultMatchItem = (
+  { label, aliases = [] }: BaseComboboxItemWithEditor,
+  query: string
+) => [label, ...aliases].some((alias) => matchWords(alias, query));
+
+interface InlineComboboxProps<TItem extends BaseComboboxItemWithEditor> {
+  trigger: string;
+  items: TItem[];
+  matchItem?: (item: TItem, query: string) => boolean;
+  renderItem?: (item: TItem) => ReactNode;
+  renderEmpty?: ReactNode;
+  onSelectItem?: (item: TItem) => void;
+}
+
+export const InlineCombobox = <TItem extends BaseComboboxItemWithEditor>({
```

Comment:
I thought about this. My concern with refactoring logic out of `inline-combobox` is I'm not sure what to do with two pieces of logic in particular: the `onCancelInput` option that requires knowledge of the trigger and the current value, and the auto-resizing input trick.

I think the `onCancelInput` is slightly too much duplication to put in multiple components, since we would risk losing consistency. If we refactored it into a package, this would make it harder for devs to customise this behaviour. It's also coupled to the concept of a "trigger" combobox, which I'm trying to keep separate from generic combobox helpers like `useComboboxInput`.

The problem with the auto-resizing input trick is it will only work reliably when paired with a specific combobox implementation. For other implementations, its styles or approach may need to be adjusted slightly. It's also too much complexity to duplicate, so I think keeping all the Ariakit combobox stuff in one place, with the auto-resizing input logic along with it, is the best option.

> it's limited by its props as the previous combobox

I'm not sure this is a problem. Devs can extend their app's inline combobox with additional props to suit one-off use cases in their app, or customise the combobox logic with changes that should apply accross all comboboxes. If they want to migrate from Ariakit to cmdk, they can make this change in one place without breaking the API in components that use the combobox.

Reply:
That makes sense, we can't package it since it's coupled to Ariakit. I've left a few suggestions to extract the content part.